agent/agent.go

```diff
@@ -393,6 +391,8 @@ func (a *Agent) finishTask(ctx context.Context, tc *taskContext, status string)
 		return nil, nil
 	}
 
+	a.killProcs(tc, false, false)
```

Comment:
This should still be in a defer so that it runs in case of a panic

Reply:
Added a defer.

Reply:
I meant that killProcs should be in a defer. There already is a panic handler for this function lower in the stack so you don't need a panic handler

Reply:
Could we put killProcs in the panic handler instead?

Reply:
I re-added the defer killProcs to where it was originally in runTask.

Reply:
Don't you call it twice now? If what you're trying to avoid is killing procs after the logger is closed in FinishTask, I think you can add your own panic handler with continue-on-error behavior in that function

Reply:
It's absolutely safe to call twice.

Reply:
My worry was that it's inefficient to call twice since we shell out? If you're not concerned about that though lgtm

Reply:
Yeah, I think it's fine. From looking at some task logs it looks very fast.